FTPizza/Ftp.cs

```diff
@@ -157,9 +163,9 @@ private string ParseItem(string item)
                     throw new Exception("Malformatted file: " + item);
                 }
 
-                while ((line = reader.ReadLine()) != null)
-                {
-                    if (line.Contains(item))
+                foreach (string file in currentDirFiles)
```

Comment:
Instead of foreaching the loop, you can just call currentDirFiles.contains(item)